### PR TITLE
Don't require consecutive Node IDs

### LINKED ISSUE #694: numbering of nodes
Maybe this comment is related to the issue "stability of node and edge IDs". 

I try to change my basis model following from the lumping tool within a python script to avoid changing my model manually in Qgis each time my basis model is changing with newer versions of the lumping/volume tools.

One of my actions is to remove isolated basin nodes (not connected to other nodes). However, when you remove a node from your model.node.static geodataframe the model is not valid anymore. The validation test is based on the number of nodes and a continues numbering of nodes, which means that no numbers may be skipped. If I would rearrange my numbering of the nodes, my edges definitions are no longer correct as well. Is it possible to keep the numbering as is, also when nodes are removed?

### PR DESCRIPTION
**Pull request by visr**
Fixes #694

It basically was already fixed in the metagraphs refactor, I just had to remove the validation check from Python.

This also adds support for 0, but not negative IDs, as I feel those could be confusing.

I used the trivial model to add some ID variations, and directly added some tests of the Arrow file results that we didn't really have yet.

### CODE REVIEW

File: core/src/bmi.jl [hunk 1]
```diff
@@ -248,7 +248,7 @@ function create_callbacks(
             save_subgrid_level,
             saved_subgrid_level;
             saveat,
-            save_start = false,
+            save_start = true,
```
Comment (visr) [inline core/src/bmi.jl:251]: @Huite I'm now saving initial subgrid levels as well, just like the regular basin.arrow. I assume you copied this from flow, which is undefined at t0 so we don't save it.

File: core/test/run_models_test.jl [hunk 2]
```diff
@@ -1,15 +1,89 @@
...
+        @test flow.to_node_id[1:4] == [6, typemax(Int), typemax(Int), 0]
+
+        @test basin.storage[1] == 1.0
+        @test basin.level[1] ≈ 0.044711584
+
+        # The exporter interpolates 1:1 for three subgrid elements, but shifted by 1.0 meter.
+        @test length(p.subgrid.level) == 3
+        @test diff(p.subgrid.level) ≈ [-1.0, 2.0]
+        # TODO The original subgrid IDs are lost and mapped to 1, 2, 3
+        @test subgrid.subgrid_id[1:3] == [11, 22, 33] broken = true
```
Comment (visr) [inline core/test/run_models_test.jl:83]: @Huite I assume you want to keep the original subgrid IDs right? If so let's create an issue to fix this. We'd have to store them in the `struct Subgrid`.

**Review by Hofer-Julian — APPROVED** [review]

File: core/src/bmi.jl (hunk 1 above)
Comment (Huite) [reply core/src/bmi.jl:251]: Yes, this sounds fine.

File: core/test/run_models_test.jl (hunk 2 above)
Comment (Huite) [reply core/test/run_models_test.jl:83]: Yes, I think we should keep them around.

Comment (visr) [reply core/test/run_models_test.jl:83]: Ok, I opened #895

**Pull request merged**